File: examples/python/testing-patterns/complex-workflows/callback-timeouts-workflow.py
from aws_durable_execution_sdk_python.config import CallbackConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    config = CallbackConfig(timeout_seconds=60, heartbeat_timeout_seconds=30)
    callback = context.create_callback(name="approval_callback", config=config)
    return f"Callback created: {callback.callback_id}"

File: examples/java/testing-patterns/complex-workflows/callback-timeouts-workflow.java
// Callback timeouts workflow handler
public class CallbackTimeoutHandler extends DurableHandler<Map, String> {
    @Override
    public String handleRequest(Map input, DurableContext ctx) {
        var config = CallbackConfig.builder()
            .timeout(Duration.ofSeconds(60))
            .heartbeatTimeout(Duration.ofSeconds(30))
            .build();

        var callback = ctx.createCallback("approval_callback", String.class, config);
        return "Callback created: " + callback.getCallbackId();
    }
}
